### PR TITLE
Add more and consistent debug logging on JMS Received/sent messages

### PR DESCRIPTION
**Pull request by jkosternl**
As requested, add more logging for the response messages. Added it for the debug log mode.

### CODE REVIEW

**Comment by alisihab** [comment]
Would it include the JMSProperties that will be sent to the target destination in the log?

File: core/src/main/java/nl/nn/adapterframework/jms/JmsSender.java [hunk 1]
```diff
@@ -245,22 +239,28 @@ public Message sendMessage(Message message, PipeLineSession session, String soap
...
 								session.put(jmsProperty, rawReplyMsg.getObjectProperty(jmsProperty));
+								if (log.isDebugEnabled()) {
+									receivedJMSProperties.append(jmsProperty).append(": ").append(rawReplyMsg.getObjectProperty(jmsProperty)).append("; ");
+								}
 							}
 						}
 					}
-					return extractMessage(rawReplyMsg, session, isSoap(), getReplySoapHeaderSessionKey(),soapWrapper);
+					logMessageDetails(rawReplyMsg);
+					log.debug("Received properties: {}", receivedJMSProperties);
```
Comment (jkosternl) [inline core/src/main/java/nl/nn/adapterframework/jms/JmsSender.java:256]: @alisihab here you can see the debug logging for the JMS Properties. It is build up at line 250. 
Are you able to test a dev-build to see if it is sufficient?

Comment (alisihab) [reply core/src/main/java/nl/nn/adapterframework/jms/JmsSender.java:256]: 'Received properties' sounds a bit confusing. It made me think it is the properties in the incoming message. I think the variable receivedJMSProperties should be renamed to something more descriptive for this specific case. I can only test when there is a new snapshot in nexus and that will probably take time to test.

Comment (jkosternl) [reply core/src/main/java/nl/nn/adapterframework/jms/JmsSender.java:256]: @alisihab, inside the `JMSSender` class and the method `sendMessage` there is a message to be sent: that's logged inside `JMSFacade#send(javax.jms.MessageProducer, javax.jms.Message, boolean)` already. 

The change here at hand, is the waiting of the synchronous response message. That's retrieved at line 238: `javax.jms.Message rawReplyMsg = mc.receive(getReplyTimeout());`
When it is received, all the JMS Properties of the received messaged is now logged as well (in debug mode). Perhaps that's a bit more than the scope of your question, but might be handy too. 
I'll figure out how to get you a new snapshot jar.

File: core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java [hunk 2]
```diff
@@ -659,6 +650,33 @@ public String send(MessageProducer messageProducer, javax.jms.Message message, b
...
+						+ "] msgID: [" + message.getJMSMessageID()
+						+ "] correlationID: [" + message.getJMSCorrelationID()
+						+ "] using deliveryMode: [" + message.getJMSDeliveryMode() + "] "
+						+ (message.getJMSReplyTo() != null ? "replyTo: [" + message.getJMSReplyTo() + "]" : ""));
+				return;
+		}
+		if (!log.isDebugEnabled()) {
+			return;
+		}
+
```
Comment (tnleeuw) [inline core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java:666]: I think that I would find it more logical to not have 2x checking `!log.isDebugEnabled()` and here an early return, but instead structure the conditions like
```
if (log.isDebugEnabled()) {
    ...
} else if (log.isInfoEnabled()) {
   ...
}
```

A matter of personal style preference perhaps so not a big thing.

**Review by tnleeuw — APPROVED** [review]

File: core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java [hunk 3]
```diff
@@ -659,6 +650,33 @@ public String send(MessageProducer messageProducer, javax.jms.Message message, b
 		}
 	}
 
+	protected void logMessageDetails(javax.jms.Message message, MessageProducer messageProducer) throws JMSException {
+		if (log.isInfoEnabled() && !log.isDebugEnabled()) {
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java:654]:
```suggestion
		if (log.isInfoEnabled()) {
```
Beetje dubbel op dit :)

File: core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java [hunk 4]
```diff
@@ -659,6 +650,33 @@ public String send(MessageProducer messageProducer, javax.jms.Message message, b
...
+
+		log.debug(getLogPrefix() + "sender on [" + getDestinationName()
+				+ "] JMSDeliveryMode=[" + message.getJMSDeliveryMode()
+				+ "] \n  JMSMessageID=[" + message.getJMSMessageID()
+				+ "] \n  JMSCorrelationID=[" + message.getJMSCorrelationID()
+				+ "] \n  JMSTimestamp=[" + DateUtils.format(message.getJMSTimestamp())
+				+ "] \n  JMSExpiration=[" + message.getJMSExpiration()
+				+ (message.getJMSPriority() != 0 ? "] \n  JMSPriority=[" + message.getJMSPriority() : "")
+				+ (message.getJMSType() != null ? "] \n  JMSType=[" + message.getJMSType() : "")
+				+ (message.getJMSReplyTo() != null ? "] \n  JMSReplyTo=[" + message.getJMSReplyTo() : "")
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java:675]: Wij gebruiken nergens (anders) newlines in de logging dus misschien dat het verstandig is het hier gelijk te trekken (en dus weg te halen).
Ook mogen deze waardes gelogd worden wanneer ze 0 zijn :)

File: core/src/main/java/nl/nn/adapterframework/jms/JmsSender.java [hunk 5]
```diff
@@ -300,7 +306,7 @@ private void setProperties(javax.jms.Message msg, ParameterValueList pvl) throws
 
 			if ((!isSoap() || !name.equals(getSoapHeaderParam()) && (StringUtils.isEmpty(getDestinationParam()) || !name.equals(getDestinationParam())))) {
 
-				if (log.isDebugEnabled()) { log.debug(getLogPrefix()+"setting ["+type+"] property from param ["+name+"] to value ["+property.getValue()+"]"); }
+				if (log.isDebugEnabled()) { log.debug("{} setting [{}] property from param [{}] to value [{}]", getLogPrefix(), type, name, property.getValue()); }
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jms/JmsSender.java:309]:
```suggestion
				log.debug("{} setting [{}] property from param [{}] to value [{}]", this::getLogPrefix, ()->type, ()->name, property::getValue);
```

Schoont aardig op zo. De logPrefix bevat niet veel extra info meer, mocht je een idee hebben om de naam op een mooie manier in de mdc te krijgen oid hoor ik het graag!

**Review by nielsm5 — CHANGES_REQUESTED** [review]

File: core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java (hunk 2 above)
Comment (jkosternl) [reply core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java:666]: I've changed it to: 
```
if (!log.isDebugEnabled()) {
   ....
   return;
}

log.debug(....).
```
assuming that info-logging is turned on mostly.

File: core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java (hunk 4 above)
Comment (jkosternl) [reply core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java:675]: Oke, dat wist ik niet. Ik had het gewoon laten staan. Ik ruim de \n nu op. De JMSType en ReplyTo zijn niet altijd gevuld.. die houd ik liever wel conditioneel, om niet te veel zut te loggen.

File: core/src/main/java/nl/nn/adapterframework/jms/JmsSender.java (hunk 5 above)
Comment (jkosternl) [reply core/src/main/java/nl/nn/adapterframework/jms/JmsSender.java]: Suggestie overgenomen. MDC-ideetje parkeer ik effe.

File: core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java [hunk 6]
```diff
@@ -659,6 +650,30 @@ public String send(MessageProducer messageProducer, javax.jms.Message message, b
...
+					+ "] JMSReplyTo=[" + message.getJMSReplyTo()
+					+ "] Message=[" + message
+					+ "]");
+		} else if (log.isInfoEnabled()){
+			log.info("[" + getName()
+					+ (messageProducer != null ? " ] message destination: [" + messageProducer.getDestination() + "] " : "")
+					+ "] msgID: [" + message.getJMSMessageID()
+					+ "] correlationID: [" + message.getJMSCorrelationID()
+					+ "] using deliveryMode: [" + message.getJMSDeliveryMode()
+					+ "] replyTo: [" + message.getJMSReplyTo() + "]");
```
Comment (nielsm5) [inline core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java:673]:
```suggestion
					+ (messageProducer != null ? " ] message destination: [" + messageProducer.getDestination() + "] " : "")
					+ "] msgID=[" + message.getJMSMessageID()
					+ "] correlationID=[" + message.getJMSCorrelationID()
					+ "] using deliveryMode=[" + message.getJMSDeliveryMode()
					+ "] replyTo=[" + message.getJMSReplyTo() + "]");
```

File: core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java (hunk 3 above)
Comment (nielsm5) [reply core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java]: Duidelijk!

File: core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java (hunk 6 above)
Comment (jkosternl) [reply core/src/main/java/nl/nn/adapterframework/jms/JMSFacade.java:673]: Ik heb de volgorde, syntax en whitespaces nu gelijk getrokken. Hopelijk kan er nu een strikje omheen ;-)

**Review by nielsm5 — APPROVED** [review]

**Pull request merged**